src/ed25519.rs

```diff
@@ -885,6 +900,12 @@ impl PublicKey {
     }
 }
 
+impl<'a> From<&'a ExpandedSecretKey> for PublicKey {
```

Comment:
This should just be `impl From<ExpandedSecretKey> for PublicKey`; using a borrow messes up the derived `Into` implementation.

And the borrow doesn't provide any benefit: it will almost certainly be eliminated by inlining, and even if it weren't, the worst case is a 32-byte stack copy, which is irrelevant compared to the cost of the basepoint multiplication.

Reply:
Fixed.